internal/core/metadata/operators/device_service/update.go

```diff
@@ -147,5 +158,47 @@ func (op deviceServiceAdminStateUpdateByName) Execute() error {
 		return err
 	}
 
+	go adminStateCallback(ds, op.lc)
+
 	return nil
 }
+
+func adminStateCallback(
+	service contract.DeviceService,
+	lc logger.LoggingClient) {
+
+	if len(service.Addressable.GetCallbackURL()) == 0 {
+		return
+	}
+
+	req, err := createCallbackRequest(service)
+	if err != nil {
+		lc.Error(fmt.Sprintf("fail to create callback request for %s", service.Name))
+	}
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		lc.Error(fmt.Sprintf("fail to invoke callback for %s, %v", service.Name, err))
+	} else if resp.StatusCode != http.StatusOK {
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			lc.Error(fmt.Sprintf("fail to read response body, %v", err))
+		}
+		lc.Error(fmt.Sprintf("fail to invoke callback for %s, %s", service.Name, string(b)))
+	}
+	resp.Body.Close()
+	resp.Close = true
+}
+
+func createCallbackRequest(service contract.DeviceService) (*http.Request, error) {
```

Comment:
add an argument in the function signature for the http method.  The http method should be hard coded here.

Reply:
Added the argument httpMethod.